client/my-sites/controller.js

```diff
@@ -336,6 +346,11 @@ export function noSite( context, next ) {
  * Set up site selection based on last URL param and/or handle no-sites error cases
  */
 export function siteSelection( context, next ) {
+	if ( isJetpackCloud() ) {
+		cloudSiteSelection( context, next );
```

Comment:
Nit, but we might as well combine these 2 lines to `return cloudSiteSelection...`

Reply:
`return cloudSiteSelection()` suggests that the function has a return value, but it doesn't. I usually prefer two separate statements. That makes it more clear that the code is procedural.